### DISCUSSION TITLE
Release 3.0.0 - Layout Editor MVP

### DISCUSSION THREAD

**Discussion by johnrm174** [johnrm174/model-railway-signalling#35] [General]
Initial release of MVP Layout Editor functionality to enable signalling schemes to be created and modified without the need to write code (all configuration is done via the UI).

What's supported in  Version 3.0.0:
- Schematic editing - points, signals and lines
- SPROG interface - DCC control of signals and points
- Basic interlocking of signals and points
- Save and load of layout schematic and state

What's coming next:
- Interlocking - of distant signals with the home signals ahead
- Schematic editing - Track sections and block instruments
- Automation - sensor events and integration with track sections and signals
- Application documentation
- MQTT networking

Any bug reports and feedback you may have would be gratefully appreciated - specifically:
- What aspects are intuitive? What aspects aren't?
- What aspects do you particularly like?
- What aspects particularly irritate you?
- what features would you like to see next?

<hr /><em>This discussion was created from the release <a href='https://github.com/johnrm174/model-railway-signalling/releases/tag/v3.0.0'>Release 3.0.0 - Layout Editor MVP</a>.</em>

**Comment by Doubletop12** [comment]
This project looks very interesting and may be of use to me. I am considering using it in a scheme to manage a large scale (7.25" gauge) railway that is over 2kms long. The plan being to monitor all the points and control the signals remotely, by radio link, back to the main control. I already have prototype radio units with track monitoring and signal outputs talking to a Raspberry Pi.

I'll admit that track control and sigmalling are new to me and I am trying to work out how I could best represent the track layout in model-railway-signalling. To all intents and purposes the track is a figure 8 with the top and bottom loops joined making one large oval. the crossover is two Y's with the station being at the joint of the two lower legs of the Y's. It is a single track with bi-directional running. The intention being that the track sensors will detect a train entering a section and set the signals at the other end to red until the section is cleared.  

The problem is I can't represent the Y as the points cannot be set vertically on the layout editor nor can I represent the out and return loops, although that may be less of a problem. I do acknowledge this layout may not be representative of full size but I'm sure being able to do it would be useful to some.

Pete

**Reply by johnrm174** [reply]
Hi Pete - thanks for the feedback. At the moment I've focussed on the "signal box diagram" type representation of track layouts. Hasn't really thought of being able to set points (and signals for that matter) vertically. That said, the underlying library functions may well support this with a few tweaks (the only faff may be offsetting the button positions). Y points may also be something to consider - added to the list of future reqs.
